### ISSUE TITLE
err:XS0091 isn't quite right

### ISSUE THREAD

**Issue by ndw** [xproc/3.0-specification#459]
Maybe we need to say:
1. All static options and variables must precede all dynamic options and variables
2. Only static options and variables may precede input/output
3. Dynamic options must precede p:declare-step and the subpipeline
4. Dynamic variables must be part of the subpipeline

**Comment by ndw** [comment]
I think the rules for static/not-static variables and options can be expressed much more simply. First, observe that you cannot have two options with the same name, so option shadowing cannot occur. If we say that it is a static error if a variable that precedes the subpipeline shadows an option or variable, then I think we're done. With those two rules, we're guaranteed that the options and variables that occur in the prologue will never shadow one another. At that point, we can stop caring about what order they occur in.

That means this

```xml
<p:declare-step type="x:foo">
  <p:option name="static" select="'text'"/>
  <p:option name="notStatic" required="true"/>
  <p:output port="result" serialization="map { 'method': $static }'"/>
  ...
</p:declare-step>
```

is perfectly fine. The `$notStatic` value isn't in-scope for the static expressions, but that's ok. And you can't declare anything that shadows it, so it can *never* be in-scope for any static expression in the prologue.

**Comment by ndw** [comment]
Closed by #462
